packages/compartment-mapper/src/bundle-json.js

```diff
@@ -0,0 +1,23 @@
+/* Provides JSON support for `bundle.js`. */
+
+const textDecoder = new TextDecoder();
+
+export default {
+  runtime: '',
+  getBundlerKit({ index, indexedImports, bytes }) {
+    const json = textDecoder.decode(bytes).trim();
+    return {
+      getFunctor: () => `\
+// === functors[${index}] ===
+(set) => set(${json}),
+`,
+      getCells: () => `\
+    { default: cell('default') },
+`,
+      getReexportsWiring: () => '',
+      getFunctorCall: () => `\
+  functors[${index}](cells[${index}].default.set);
```

Comment:
grepping for these four method names, I only see uses with bundle.js and a definition of a bundle with three of them. Where can I find an explanation of what they mean?

Reply:
There is no explanation except how they’re used in bundle.js.

Reply:
I'm wondering if it'd make sense to replace some of this with empty strings or no-ops and inline json in the cell

```
  getCells: () => `\
    { default: cell('default', ${json}) },
`,
```

Reply:
Oh, no. That doesn’t work because `cell` and `cells` are not in the lexical scope of functors.

Reply:
Oh, they were in scope! I’ve added a commit that adds docs, asserts this invariant, and fixes it. There should not be any lexical variables in scope that might mask a global.